application/backend/app/api/routers/sinks.py

```diff
@@ -154,12 +164,22 @@ def update_sink(
         ),
     ],
     configuration_service: Annotated[ConfigurationService, Depends(get_configuration_service)],
-) -> Sink:
+) -> SinkView:
     """Reconfigure an existing sink"""
     if "sink_type" in sink_config:
         raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="The 'sink_type' field cannot be changed")
     try:
-        return configuration_service.update_sink(sink_id, sink_config)
+        sink = configuration_service.get_sink_by_id(sink_id)
+        updated_sink: Sink = sink.model_copy(update=sink_config)
+        updated_sink.config_data = updated_sink.config_data.model_copy(update=sink_config)
```

Comment:
1. Shall we introduce a separate schema for update operations? Now, the client can request any payload in `PATCH` and it will be silently ignored, as `model_copy` would not complain. 
2. We check resource existence twice now: on service layer and here in routes. This can be probably done only once on the service layer.